### ISSUE TITLE
Incorrect TextInput email field suggestions (contacts, Hide my email)

### ISSUE THREAD

**Issue by troyshu** [facebook/react-native#46022]
### Description

Not sure if this issue should go in the expo or react-native repo, but posting it here.

When entering text into an email TextField (`textContentType="emailAddress"`), two bugs with incorrect suggestions occur on when the app is run on a **plugged in iOS device**:

**Bug 1**: "Hide my Email" is suggested twice. Selecting the first "Hide my email" incorrectly inserts the string "Hide my email" into the text field.

**Bug 2**: an email address from Contacts not being suggested when the email is partially typed (expected behavior from Apple's SwiftRead developer docs). An email address is only suggested when the contact's first/last name is partially typed.

See attached screen recording for demo of the two bugs.

Note: only the second bug occurs in an iOS simulator because the first bug requires the user to be logged into iCloud for the "Hide my Email" functionality

### Steps to reproduce

1. Clone https://github.com/troyshu/expo-text-input-test
2. Plug in iOS device, enable USB debugging
3. `npm install` then `npm run ios:device` to run on device
4. Try to enter an email in the email field. You should see Bug 1 and Bug 2 as defined above.
 

### React Native Version

0.74.5

### Affected Platforms

Runtime - iOS

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.5
  CPU: (12) arm64 Apple M2 Max
  Memory: 103.22 MB / 64.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 21.5.0
    path: /opt/homebrew/bin/node
  Yarn:
    version: 1.22.21
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.2.4
    path: /opt/homebrew/bin/npm
  Watchman:
    version: 2023.12.04.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.14.3
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.5
      - iOS 17.5
      - macOS 14.5
      - tvOS 17.5
      - visionOS 1.2
      - watchOS 10.5
  Android SDK:
    API Levels:
      - "28"
      - "29"
      - "30"
      - "31"
      - "33"
      - "34"
    Build Tools:
      - 29.0.2
      - 30.0.2
      - 30.0.3
      - 33.0.0
      - 33.0.1
      - 34.0.0
    System Images:
      - android-30 | Google APIs Intel x86 Atom
      - android-30 | Google Play Intel x86 Atom
      - android-30 | Google Play Intel x86 Atom_64
      - android-31 | Google APIs Intel x86 Atom_64
      - android-34 | Google Play ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2023.1 AI-231.9392.1.2311.11330709
  Xcode:
    version: 15.4/15F31d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.4.1
    path: /usr/bin/javac
  Ruby:
    version: 3.2.2
    path: /opt/homebrew/opt/ruby/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.5
    wanted: 0.74.5
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: true
  newArchEnabled: false
```
```


### Stacktrace or Logs

```text
NA
```

### Reproducer

https://github.com/troyshu/expo-text-input-test

### Screenshots and Videos

https://github.com/user-attachments/assets/98e22f52-d6a2-4a49-bc11-8ea8f0e715d3

**Comment by IkeyBenz** [comment]
I'm noticing this as well.

**Comment by troyshu** [comment]
(pinging @cortinico just because they're the only RN contributor on this issue) Any thoughts from the RN team about this issue? Having two "Hide my email" suggestions over an email input field, and one of them inserting the string "Hide my email" into the email input when tapped, is confusing for users and leaves an app feeling unpolished for them.

**Comment by cortinico** [comment]
> ###
> Not sure if this issue should go in the expo or react-native repo, but posting it here.

Have you opened an issue on Expo as well by any chance?

**Comment by troyshu** [comment]
Thanks for responding @cortinico. I hadn't, but I just posted the issue to expo's repo as well: https://github.com/expo/expo/issues/32483. Correct me if I'm wrong, but since this issue is with a react-native component, (`import { TextInput } from "react-native"`), shouldn't it be looked at here?

**Comment by cortinico** [comment]
> Thanks for responding @cortinico. I hadn't, but I just posted the issue to expo's repo as well: [expo/expo#32483](https://github.com/expo/expo/issues/32483). Correct me if I'm wrong, but since this issue is with a react-native component, (`import { TextInput } from "react-native"`), shouldn't it be looked at here?

Yes you're right, but our team is extremely small and doesn't have capacity to look into any issue that gets reported (we get a lot of them). As you're using Expo, having the issue opened also in Expo will help to exclude any problems which might be Expo specific

**Comment by troyshu** [comment]
Totally understand. Thanks for all the amazing work that you and the RN team do! Hope someone can take a look at this soon.

**Comment by brentvatne** [comment]
@cortinico - this shouldn't be related to expo, we don't interfere with core apis from react-native like TextInput

**Comment by brentvatne** [comment]
@troyshu - can you try updating to 0.76 and seeing if that resolves your issue? https://expo.dev/changelog/2024/10-24-sdk-52-beta - you can run `npx expo install expo@next --fix` in your repro project

**Comment by troyshu** [comment]
I ran `npx expo install expo@next --fix` to update to Expo 52 beta and React Native 0.76, still seeing the same issue on my iOS device.

**Comment by cortinico** [comment]
cc @cipolleschi if you have any insights on what could be causing this on iOS

**Comment by cipolleschi** [comment]
@troyshu is this happening also if you opt out from the New Architecture? Just to understand whether it has always been like that or not. I don't see anything in the codebase that suggests that we add the text twice.

**Comment by migueldaipre** [comment]
This is probably related to something native on  the apple side. Flutter has the same issue. As can be seen [here](https://github.com/flutter/flutter/issues/157495)

**Comment by coado** [comment]
Does the issue depend on the iOS version? I've checked on 17.6.1 and there is only one "Hide my email" which works as expected.

**Comment by jmallian** [comment]
This is not a react native or expo issue. I came across this thread because I'm having the same issue with native iOS, xcode 16.1, iOS 18.1, with the following swift code: 

> TextField("Email", text: $email)
>                     .textFieldStyle(RoundedBorderTextFieldStyle())
>                     .textContentType(.emailAddress)
>                     .keyboardType(.emailAddress)
>                     .autocorrectionDisabled()

 Ignoring the fact that it shows two Hide My Email options, what I actually want is for it to suggest my actual email.
